portal/src/main/webapp/WEB-INF/jsp/visualize.jsp

```diff
@@ -495,7 +503,11 @@ window.loadReactApp({ defaultRoute: 'results' });
         );
 
         //Move code related to expression tab from cross_cancer_results.jsp to here
-        window.ccQueriedGenes = OQL.genes(window.serverVars.theQuery);
+        window.ccQueriedGenes;
+        var isGenesetQuery = jspToJs('<%=isGenesetQuery%>');
+        if (!isGenesetQuery) {
+            window.ccQueriedGenes = OQL.genes(window.serverVars.theQuery);
+        }
```

Comment:
@adamabeshouse I haven't introduced this OQL line here, I just added a check so that this line is skipped when having queries with only gene sets. This was added in #3071, which you approved. Maybe you can recheck it and then we decide what to do here.

Reply:
like I said, lines 506+ are good. I was talking about this: https://github.com/cBioPortal/cbioportal/pull/3940/files#diff-5c6909f939f5f14fdf689eaac83ae729R78